src/Core/Context/CurrentContext.cs

```diff
@@ -184,6 +190,7 @@ private List<CurrentContextOrganization> GetOrganizations(Dictionary<string, IEn
                     Id = new Guid(c.Value),
                     Type = OrganizationUserType.Owner,
                     AccessSecretsManager = accessSecretsManager.ContainsKey(c.Value),
+                    LimitCollectionCreationDeletion = !cachedOrgs.ContainsKey(new Guid(c.Value)) || cachedOrgs[new Guid(c.Value)].LimitCollectionCreationDeletion,
```

Comment:
This might be a little more efficient because we're only accessing the dictionary once
```suggestion
                    LimitCollectionCreationDeletion = cachedOrgs[new Guid(c.Value)]?.LimitCollectionCreationDeletion ?? false,
```